pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamConfig.java

```diff
@@ -197,6 +199,19 @@ public StreamConfig(Map<String, String> streamConfigMap) {
       _flushSegmentDesiredSizeBytes = DEFAULT_DESIRED_SEGMENT_SIZE_BYTES;
     }
 
+    int initialRowsThreshold = 0;
+    String initialRowsThresholdValue = streamConfigMap.get(StreamConfigProperties.SEGMENT_INITIAL_ROWS_THRESHOLD);
+    if (initialRowsThresholdValue != null) {
+      try {
+        initialRowsThreshold = Integer.parseInt(initialRowsThresholdValue);
```

Comment:
You may want to check if it is at least some threshold value? Or positive value?

Reply:
I've checked for positive, 2 lines below this. I dont think we should check for some minimum value... it might backfire in some cases if threshold was actually expected to be small